Improve contributing guide (#53)
lightspeed/flame: .github/CONTRIBUTING.md

# Contributing

Lightspeed Flame is a project that is actively used and maintained to build Lightspeed products internally. It's being treated the same way as any internal development work and follows our product roadmap as with any features built within Lightspeed.

The main purpose of working on it open-source is to be transparent on what we bring to our customers, but also to be able to get valuable feedback and contribution from the community. We believe open-source is vital to software development as most of the libraries we rely on are built in the open space.

> As our journey begins in the open-source world, we are accepting [bug reports](https://github.com/lightspeed/flame/issues/new?template=BUG_REPORT.md), [feature request](https://github.com/lightspeed/flame/issues/new?template=FEATURE_REQUEST.md) as well as pull requests for improvements, but no additions or breaking changes until we have a proper process to accept and treat those accordingly.

Reviewer: ```suggestion
As our journey begins in the open-source world, we are accepting [bug reports](https://github.com/lightspeed/flame/issues/new?template=BUG_REPORT.md), [feature requests](https://github.com/lightspeed/flame/issues/new?template=FEATURE_REQUEST.md) and pull requests for improvements. We are not accepting no additions or breaking changes until we have a proper process to triage and approve these changes accordingly.
```

> We still encourage anyone to fork and play around Flame for personal usage. Read our [LICENSE](../LICENSE) in case of uncertainty on what we allow in this regard.

Reviewer: The second sentence is a little vague, but I guess the license only pertains to users who would like to use Flame commercially. If I'm correct, consider the following suggestion:

```suggestion
We still encourage anyone to fork and play around Flame for personal usage. Read our [LICENSE](../LICENSE) in the case of commercial use.
```

To create a brand new component, use the following command that will create a base skeleton ready for development:

```sh
yarn generate
```

You will be given a choice of "Mainline" or "Add-on" component, then have to choose a name for the component.

#### Mainline component

> Mainline components will be created under `packages/flame/src/{ComponentName}`, where all other components are located. This will create the base component structure ready for development, including README for docs, story for Storybook development and visual regression test, tests and TypeScript component file.

Reviewer: Usually, when we create a comma separated list, we aim to have three separate items that are fairly similar in length. The flow helps with readability. Consider the following suggestion:

```suggestion
Mainline components will be created under `packages/flame/src/{ComponentName}`, where all other components are located. The base component structure includes a storybook file, a test file, and a component file.
```

#### Add-on component

> Add-on component will be created under the `packages/flame-{component-name}`. It should be used for add-on/experimental components that are not ready to go into the [mainline package](https://github.com/lightspeed/flame/tree/master/packages/flame) or are considered as "extras" and don't fit our component library.

Reviewer: Restructured to explain what is defined as an "add-on" component:

```suggestion
Add-on components will be created under the `packages/flame-{component-name}`. These components are either experimental in implementation, or compositions that do not fit our essential component library.
```

> Add-on components are a completely separate package with their own version and have `@lightspeed/flame` and all its required dependencies as `peerDependencies`. They include the same base component structure as a mainline component, but also have a `tsconfig` to output types as well as their own LICENSE and CHANGELOG files.

Reviewer: I don't think we need to be too granular in the component structure (re: CHANGELOG and LICENSE) if you word this section as follows:

```suggestion
Add-on components are a completely separate package which are versioned independently, licensed independently, and have `@lightspeed/flame` and all its required dependencies as `peerDependencies`. They include the same base component structure as a mainline component.
```

Independent versioning and licensing implies that the package has its own LICENSE and CHANGELOG.
